src/api/ledger/transactions.js

```diff
@@ -62,24 +62,45 @@ function getAccountTx(remote, address, options, marker, limit, callback) {
   });
 }
 
-function getTransactionsInternal(remote, address, options, callback) {
+function checkForLedgerGaps(remote, options, transactions) {
+  let {minLedgerVersion, maxLedgerVersion} = options;
+
+  // if we reached the limit on number of transactions, then we can shrink
+  // the required ledger range to only guarantee that there are no gaps in
+  // the range of ledgers spanned by those transactions
+  if (options.limit && transactions.length === options.limit) {
```

Comment:
Are there any use cases for this where this behaviour would be potentially dangerous that you know of? This seems OK to me, but I feel like I'm missing something.